### DISCUSSION TITLE
Intermittent Restart Manager Errors in v3.5

### DISCUSSION THREAD

**Discussion by tdbuchanan** [ravendb/ravendb#13065] [Q&A]
I've been running a version 3.5 database (now on patch 35311) for a long time, and until recently it has worked flawlessly.  Over the past few months there's been an increase in errors stemming from Restart Manager actions triggered by index maintenance processes.  The follow snippet is representative of what I'm seeing on a recurring basis:

> System.ComponentModel.Win32Exception: Access is denied at Raven.Database.Extensions.WhoIsLocking.GetProcessesUsingFile(String filePath) in C:\Builds\RavenDB-3.5-Patch\Raven.Database\Extensions\WhoIsLocking.cs:line 45 at Raven.Database.Extensions.WhoIsLocking.ThisFile(String path) in C:\Builds\RavenDB-3.5-Patch\Raven.Database\Extensions\WhoIsLocking.cs:line 19 at Raven.Database.Extensions.IOExtensions.TryHandlingError(String directory, Int32 i, Exception e) in C:\Builds\RavenDB-3.5-Patch\Raven.Database\Extensions\IOExtensions.cs:line 115 at Raven.Database.Extensions.IOExtensions.DeleteDirectory(String directory) in C:\Builds\RavenDB-3.5-Patch\Raven.Database\Extensions\IOExtensions.cs:line 97 at Raven.Database.Indexing.IndexStorage.DeleteIndexData(Int32 id) in C:\Builds\RavenDB-3.5-Patch\Raven.Database\Indexing\IndexStorage.cs:line 1042 at Raven.Database.Indexing.IndexStorage.TryResettingIndex(String indexName, IndexDefinition indexDefinition, Boolean onStartup) in C:\Builds\RavenDB-3.5-Patch\Raven.Database\Indexing\IndexStorage.cs:line 386 --- End of inner exception stack trace --- at Raven.Database.Indexing.IndexStorage.TryResettingIndex(String indexName, IndexDefinition indexDefinition, Boolean onStartup) in C:\Builds\RavenDB-3.5-Patch\Raven.Database\Indexing\IndexStorage.cs:line 390 at Raven.Database.Indexing.IndexStorage.OpenIndex(String indexName, Boolean onStartup, Boolean forceFullIndexCheck) in C:\Builds\RavenDB-3.5-Patch\Raven.Database\Indexing\IndexStorage.cs:line 284

There have been no changes in the database configuration or the application logic accessing the database since well before this started happening.  I'm assuming the index file(s) are deadlocked by some other system process (antivirus??), but I'm not much of an admin and I'm not quite sure how to definitively pinpoint the problem.  Any suggestions would be greatly appreciated.

**Comment by ayende** [comment]
1. Note that this is no longer a supported version. You are _strong_ encouraged to upgrade to the current version.
2. This looks like a lock (missing context here), do you have anti virus or indexer process that may be locking the files?
